### DISCUSSION TITLE
Collapsible network graph

### DISCUSSION THREAD

**Discussion by loopmode** [airbnb/visx#1137] [Q&A]
I'm using `@vx/network` and users of the app can draw graphs of arbitrary complexity.

Is it possible to have collapsible functionality, much like with the many Tree and hierarchy examples, but with `import { Graph } from '@vx/network'`?
I've read https://github.com/hshoff/vx/issues/162 and looked at the examples and their code, and by the looks of the examples, it's just what I need, any of those :)

However, I'm not sure how to approach the problem with the network data structure. I do not necessarily have a directed graph or a single root node..
Anybody done collapsible network graphs already? I'd appreciate any idea or suggestion.

**Comment by williaster** [comment]
Hey @loopmode you might be interested in checking out `@data-ui/network` which you can demo [here](https://williaster.github.io/data-ui/?selectedKind=network&selectedStory=Animated%20expandable%20network%20with%20custom%20renderer&full=0&addons=0&stories=1&panelRight=0), example source code [here](https://github.com/williaster/data-ui/tree/master/packages/demo/examples/04-network) and implementation [here](https://github.com/williaster/data-ui/tree/master/packages/network/src).

It's partially implemented with `vx`, but not `@vx/network`. We discussed sometime porting this back to `vx` once we figured out the API around layout etc., which is much more complex than what `@vx/network` currently implements. It's definitely a "higher order chart" which feels less at home in `vx`, but perhaps it'll be useful or we could discuss how to bring back the primatives to `vx`. We've successfully used this component in a handful of projects with different layouts and custom nodes/links.

cc @conglei 
![network](https://user-images.githubusercontent.com/4496521/44805948-62fe6880-ab7a-11e8-9fd8-119c0351fb78.gif)
